src/cache/inmemory/types.ts

```diff
@@ -49,7 +49,8 @@ export interface NormalizedCache {
   merge(olderId: string, newerObject: StoreObject): void;
   merge(olderObject: StoreObject, newerId: string): void;
 
-  modify(dataId: string, fields: Modifiers | Modifier<any>): boolean;
+  modify<Entity extends Record<string, any>>(dataId: string, fields: Modifiers<Entity>): boolean;
+  modify<Entity>(dataId: string, modifier: Modifier<Entity>): boolean;
```

Comment:
We should probably use

```suggestion
  modify<Entity>(dataId: string, modifier: AllFieldsModifier<Entity>): boolean;
```

Reply:
Very good catch!